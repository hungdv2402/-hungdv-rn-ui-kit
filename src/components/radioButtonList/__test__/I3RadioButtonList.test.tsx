import renderer from 'react-test-renderer';
import I3RadioButtonList from '../I3RadioButtonList';
import React from 'react';
import { View } from 'react-native';
import SizeBox from '../../sizeBox';
import {render, fireEvent} from '@testing-library/react-native'

describe('renders I3RadioButtonList correctly', () => {
  it('should render I3RadioButtonList with basic properties', () => {
    const tree = renderer
      .create(
        <I3RadioButtonList
          data={[]}
          value={''}
          onSelect={jest.fn()}
        />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render I3RadioButtonList with custom properties and sample data', () => {
    const tree = renderer
      .create(
        <View>
          <I3RadioButtonList
            containerStyle={{ backgroundColor: 'red' }}
            labelStyle={{ fontSize: 20 }}
            itemStyle={{ padding: 20 }}
            data={[{ label: 'Option 1', id: '1' }, { label: 'Option 2', id: '2' }]}
            value={'1'}
            onSelect={jest.fn()}
          />
          <SizeBox height={10} />
        </View>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should select correct a radio button in I3RadioButtonList', async () => {
    let value = '1'
    const onSelect = jest.fn((res: string) => {value = res});
    const radioButtonList = render(
      <I3RadioButtonList
      containerStyle={{ backgroundColor: 'red' }}
      labelStyle={{ fontSize: 20 }}
      itemStyle={{ padding: 20 }}
      data={[{ label: 'Option 1', id: '1' }, { label: 'Option 2', id: '2' }]}
      value={value}
      onSelect={onSelect}
    />)
    const radioButton1 = await radioButtonList.findByTestId('radio-button-2');
    fireEvent.press(radioButton1);
    expect(value).toBe('2');
    expect(onSelect).toHaveBeenCalled();
  });
});