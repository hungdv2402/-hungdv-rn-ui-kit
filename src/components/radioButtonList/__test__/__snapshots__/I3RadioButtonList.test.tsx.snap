// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders I3RadioButtonList correctly should render I3RadioButtonList with basic properties 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexGrow": 0,
      },
      undefined,
    ]
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;

exports[`renders I3RadioButtonList correctly should render I3RadioButtonList with custom properties and sample data 1`] = `
<View>
  <RCTScrollView
    data={
      [
        {
          "id": "1",
          "label": "Option 1",
        },
        {
          "id": "2",
          "label": "Option 2",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "flexGrow": 0,
        },
        {
          "backgroundColor": "red",
        },
      ]
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "opacity": 1,
                "padding": 20,
              }
            }
            testID="radio-button-1"
          >
            <Image
              source={
                {
                  "testUri": "../../../src/components/radioButtonList/assets/selectedRadioButton.png",
                }
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
            />
            <View
              style={
                {
                  "height": undefined,
                  "width": 8,
                }
              }
            />
            <Text
              style={
                [
                  {
                    "color": "#212127",
                    "fontSize": 17,
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              Option 1
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "opacity": 1,
                "padding": 20,
              }
            }
            testID="radio-button-2"
          >
            <Image
              source={
                {
                  "testUri": "../../../src/components/radioButtonList/assets/radioButton.png",
                }
              }
              style={
                {
                  "height": 24,
                  "width": 24,
                }
              }
            />
            <View
              style={
                {
                  "height": undefined,
                  "width": 8,
                }
              }
            />
            <Text
              style={
                [
                  {
                    "color": "#212127",
                    "fontSize": 17,
                  },
                  {
                    "fontSize": 20,
                  },
                ]
              }
            >
              Option 2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      {
        "height": 10,
        "width": undefined,
      }
    }
  />
</View>
`;
